torch/distributed/remote_device.py

```diff
@@ -0,0 +1,111 @@
+from typing import Optional, Union
+
+import torch
+
+
+class _RemoteDevice(object):
+    """
+    Represents a device on a remote worker.
+
+    Args:
+        remote_device (str or int): Represents a device on a remote worker.
+            The format should be one of the following:
+
+                1. "<workername>/<device>", where the device field can be parsed as torch.device type.
+                   E.g., "trainer0/cpu", "trainer0", "ps0/cuda:0".
+                   In addition, the device field can be optional and the default value is "cpu".
+                2. "rank:<rank>/<device>", where <rank> is the rank of the
+                   process and device can be parsed as torch.device type.
+                   E.g., "rank:0/cpu", "rank:0", "rank:0/cuda:0"
+                3. <workername> and <rank> are optional and formats like "cpu"
+                    and "cuda:1", just represent local devices.
+    """
+
+    def __init__(self, remote_device: Union[str, int]):
```

Comment:
what does a `int` mean from user prospective? is it an index to local cuda device? I think we might want to accept a string or a torch.Device instead of int?

Reply:
integer is similar to torch.device, where someone can do something like `torch.device(0)` and it is cuda:0. So you can do _RemoteDevice(0) and it will actually be just local cuda:0. Although, I do think your suggestion is better where it is only str or torch.device. I'll update this.